lychee-bin/src/formatters/response/color.rs

```diff
@@ -10,52 +10,75 @@ use super::{ResponseFormatter, MAX_RESPONSE_OUTPUT_WIDTH};
             Status::Redirected(_) => &NORMAL,
             Status::UnknownStatusCode(_) | Status::Timeout(_) => &YELLOW,
             Status::Error(_) | Status::Cached(CacheStatus::Error(_)) => &PINK,
-        };
+        }
+    }
 
-        let status_formatted = format_status(&body.status);
+    /// Format the status code or text for the color formatter.
+    ///
+    /// - Numeric status codes are right-aligned.
+    /// - Textual statuses are left-aligned.
+    /// - Padding is taken into account.
+    fn format_status(status: &Status) -> String {
+        let status_code_or_text = status.code_as_string();
 
-        let colored_status = status_color.apply_to(status_formatted);
+        // Calculate the effective padding. Ensure it's non-negative to avoid panic.
+        let padding = MAX_RESPONSE_OUTPUT_WIDTH.saturating_sub(status_code_or_text.len() + 2); // +2 for brackets
 
-        // Construct the output.
-        format!("{} {}", colored_status, body.uri)
+        format!(
+            "{}[{:>width$}]",
+            " ".repeat(padding),
+            status_code_or_text,
+            width = status_code_or_text.len()
+        )
+    }
+
+    /// Color and format the response status.
+    fn format_response_status(status: &Status) -> String {
+        let status_color = ColorFormatter::status_color(status);
+        let formatted_status = ColorFormatter::format_status(status);
+        status_color.apply_to(formatted_status).to_string()
     }
 }
 
-/// Format the status code or text for the color formatter.
-///
-/// Numeric status codes are right-aligned.
-/// Textual statuses are left-aligned.
-/// Padding is taken into account.
-fn format_status(status: &Status) -> String {
-    let status_code_or_text = status.code_as_string();
-
-    // Calculate the effective padding. Ensure it's non-negative to avoid panic.
-    let padding = MAX_RESPONSE_OUTPUT_WIDTH.saturating_sub(status_code_or_text.len() + 2); // +2 for brackets
-
-    format!(
-        "{}[{:>width$}]",
-        " ".repeat(padding),
-        status_code_or_text,
-        width = status_code_or_text.len()
-    )
+impl ResponseFormatter for ColorFormatter {
+    fn format_response(&self, body: &ResponseBody) -> String {
+        let colored_status = ColorFormatter::format_response_status(&body.status);
+        format!("{} {}", colored_status, body.uri)
+    }
+
+    /// Provide some more detailed information about the response
+    /// This prints the entire response body, including the exact error message
+    /// (if available).
+    fn format_detailed_response(&self, body: &ResponseBody) -> String {
+        let colored_status = ColorFormatter::format_response_status(&body.status);
+        format!("{colored_status} {body}")
+    }
 }
 
 #[cfg(test)]
 mod tests {
     use super::*;
     use http::StatusCode;
     use lychee_lib::{ErrorKind, Status, Uri};
+    use pretty_assertions::assert_eq;
+
+    #[cfg(test)]
```

Comment:
What is the purpose of this annotation? Isn't it redundant, since the whole module `tests` is already marked as such?

Reply:
You're right. It's unnecessary.